logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,228 @@
+            new AlluxioLog4jSocketNode(this, client);
+        RetryPolicy retryPolicy = new ExponentialBackoffRetry(
+            BASE_SLEEP_TIME_IN_MS, MAX_SLEEP_TIME_IN_MS, MAX_NUM_RETRY);
+        while (true) {
+          try {
+            mThreadPool.execute(clientSocketNode);
+            break;
+          } catch (Throwable t) {
+            if (t instanceof RejectedExecutionException) {
+              if (!retryPolicy.attemptRetry()) {
+                LOG.warn("Connection with {} has been rejected by ExecutorService {} times"
+                    + "till timedout, reason: {}",
+                    inetAddress.getHostAddress(), retryPolicy.getRetryCount(), t);
+                client.close();
+                break;
+              }
+            } else if (t instanceof Error) {
+              LOG.error("ExecutorService threw error: ", t);
+              throw (Error) t;
+            } else {
+              LOG.warn("ExecutorService threw error: ", t);
+              break;
+            }
+          }
+        }
+      } catch (IOException e) {
+        if (!mStopped) {
+          LOG.warn("Socket transport error occurred during accepting message.", e);
+        }
+        break;
+      }
+    }
+  }
+
+  /**
+   * Called from another thread (actually in the shutdown hook of {@link AlluxioLogServer}
+   * to stop the main thread of {@link AlluxioLogServerProcess}.
+   */
+  private void stopServing() {
+    mStopped = true;
+    if (mServerSocket != null) {
+      try {
+        mServerSocket.close();
+      } catch (IOException e) {
+        LOG.warn("Exception in closing server socket.", e);
+      }
+    }
+
+    mThreadPool.shutdown();
+    long timeoutMS = STOP_TIMEOUT_IN_MS;
+    long now = System.currentTimeMillis();
+    while (timeoutMS >= 0) {
+      try {
+        mThreadPool.awaitTermination(timeoutMS, TimeUnit.MILLISECONDS);
+        break;
+      } catch (InterruptedException e) {
+        long newnow = System.currentTimeMillis();
+        timeoutMS -= (newnow - now);
+        now = newnow;
+      }
+    }
+  }
+
+  /**
+   * Configure a {@link Hierarchy} instance used to retrive logger by name and maintain the logger
+   * hierarchy. An instance of this class will be passed to a {@link AlluxioLog4jSocketNode} so
+   * that the {@link AlluxioLog4jSocketNode} instance can retrieve the logger to log incoming
+   * {@link org.apache.log4j.spi.LoggingEvent}s.
+   *
+   * @param inetAddress inet address of the client
+   * @param logAppenderName name of the appender to use for this client
+   * @return A {@link Hierarchy} instance to pass to {@link AlluxioLog4jSocketNode}
+   * @throws IOException if fails to create an {@link FileInputStream} to read log4j.properties
+   * @throws URISyntaxException if fails to derive a valid URI from the value of property named
+   *         "log4j.configuration"
+   */
+  protected LoggerRepository configureHierarchy(InetAddress inetAddress, String logAppenderName)
```

Comment:
This method is only used in `AlluxioLog4jSocketNode`, could we move it there?

Reply:
I think `AlluxioLogServerProcess` has the core logic of processing log-related tasks, and we pass `AlluxioLogServerProcess` instances as reference to worker threads. The worker threads call `AlluxioLogServerProcess` methods to perform certain task-related tasks. Therefore, I think it's fine to put this method in `AlluxioLogServerProcess`.

Reply:
We talked offline and decided to move the logic to `AlluxioLog4jSocketNode` to keep control flow within `AlluxioLog4jSocketNode` self-contained

Reply:
Done.